File: Test02-1_ws/src/demo_cpp_pkg/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"

class PersonNode :public rclcpp::Node
{
private:
    std::string name_;
    int age_;
public:
    PersonNode(const std::string &node_name,const std::string &name,const int &age)
    :Node(node_name) // 调用父类构造函数
    {
        this -> name_ = name;
        this -> age_ = age;
    }
    
    void eat(const std::string &foodName) // const的作用是保证函数内不修改类成员变量
    {
        RCLCPP_INFO(this->get_logger(), "我是%s,%d岁,I like eating %s", this->name_.c_str(),
        this->age_,foodName.c_str());
    };
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("person_node","Alice", 30); //auto是类型自动推导 
    node->eat("菜菜");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: Test02-1_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
#!/home/pjc/miniconda3/envs/ros2_humble/bin/python3
# 代表默认Python解释器路径
# -*- coding: utf-8 -*-

import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self, node_name:str,name:str, age:int):
        print("Person init方法被调用")
        super().__init__(node_name)
        self.name = name
        self.age = age

    def eat(self,foodName:str):
        # return f"Hello, my name is {self.name} and I am {self.age} years old,I like {foodName}."
        self.get_logger().info(f"Hello, my name is {self.name} and I am {self.age} years old,I like eat {foodName}.")

def main():
    rclpy.init()
    node = PersonNode("person_node","Alice", 30)
    node.eat("pizza")
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()
    